### PR TITLE
Improve the juju unit status

### PR DESCRIPTION
**Pull request by addyess**
# Summary
Since the charm reconcile loops runs the same execution steps over and over, providing meaning descriptions of those stages is important.  These changes better describe the stages the charm is stepping through -- not just the early stages

# Details
* Many of the MaintenanceStatus messages were update to show that they are repeating the same stage of configuration, which may already be applied
* Improved the worker's status when it's blocked by not being related to the k8s control-plane application

### CODE REVIEW

File: charms/worker/k8s/src/charm.py [hunk 1]
```diff
@@ -346,14 +341,14 @@ def _create_cos_tokens(self):
             self.distributor.allocate_tokens(relation=rel, token_strategy=TokenStrategy.COS)
 
     @on_error(
-        WaitingStatus("Waiting for enable functionalities"),
+        WaitingStatus("Waiting to enable Functionalities"),
```
Comment (mateoflorido) [inline charms/worker/k8s/src/charm.py:344]: We are now using ✨ `features` ✨ 
```suggestion
        WaitingStatus("Waiting to enable features"),
```

File: charms/worker/k8s/src/charm.py [hunk 2]
```diff
@@ -346,14 +341,14 @@ def _create_cos_tokens(self):
...
         InvalidResponseError,
         K8sdConnectionError,
     )
     def _enable_functionalities(self):
         """Enable necessary components for the Kubernetes cluster."""
-        status.add(ops.MaintenanceStatus("Enabling Functionalities"))
-        log.info("Enabling Functionalities")
+        status.add(ops.MaintenanceStatus("Updating K8s Functionalities"))
+        log.info("Enabling K8s Functionalities")
```
Comment (mateoflorido) [inline charms/worker/k8s/src/charm.py:351]:
```suggestion
        status.add(ops.MaintenanceStatus("Updating K8s features"))
        log.info("Enabling K8s features")
```

File: charms/worker/k8s/src/charm.py [hunk 3]
```diff
@@ -495,10 +490,12 @@ def _update_status(self):
             status.add(ops.WaitingStatus("Preparing to leave cluster"))
             return
         if self.is_worker:
-            relation = self.model.get_relation("cluster")
-            assert relation, "Missing cluster relation with k8s"  # nosec
+            if not self.model.get_relation("cluster"):
+                status.add(ops.BlockedStatus("Missing cluster integration"))
```
Comment (mateoflorido) [inline charms/worker/k8s/src/charm.py:494]: Love it!

**Review by mateoflorido — CHANGES_REQUESTED** [review]
I love this PR, nice work! Messages are much cleaner now. Few nits, but LGTM

**Review by mateoflorido — APPROVED** [review]
LGTM 💯

File: tests/integration/test_etcd.py [hunk 4]
```diff
@@ -40,4 +40,4 @@ async def test_etcd_datastore(kubernetes_cluster: model.Model):
     status = json.loads(result.results["stdout"])
     assert status["ready"], "Cluster isn't ready"
     assert status["datastore"]["type"] == "external", "Not bootstrapped against etcd"
-    assert status["datastore"]["external-url"] == f"https://{etcd.public_address}:{etcd_port}"
+    assert status["datastore"]["servers"] == [f"https://{etcd.public_address}:{etcd_port}"]
```
Comment (addyess) [inline tests/integration/test_etcd.py:43]: @mateoflorido 
the k8s snap changed on us now that the datastore status is a little different

Comment (mateoflorido) [reply tests/integration/test_etcd.py:43]: Yes, that changed a few days ago. Thanks for the update!

**Pull request merged**